### PR TITLE
Deprecate ``is_typing_guard`` and ``is_sys_guard``

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Description

See https://github.com/PyCQA/pylint/pull/5107#discussion_r720916886 for context.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Related Issue

References #1199

### CODE REVIEW

File: astroid/nodes/node_classes.py [hunk 1]
```diff
@@ -2870,6 +2871,10 @@ def is_sys_guard(self) -> bool:
         >>> node.is_sys_guard()
         True
         """
+        warnings.warn(
+            "The 'is_sys_guard' function is deprecated and will be removed in astroid 3.0.0",
```
[the remark this answers is not in the document]
Comment (Pierre-Sassoulas) [reply astroid/nodes/node_classes.py:2875]: Makes sense, good call.

File: ChangeLog [hunk 2]
```diff
@@ -40,6 +40,10 @@ Release date: TBA
 
   Closes PyCQA/pylint#5059
 
+* The ``is_typing_guard`` and ``is_sys_guard`` functions were deprecated and will
```
Comment (cdce8p) [inline ChangeLog:43]:
```suggestion
* The ``is_typing_guard`` and ``is_sys_guard`` functions are deprecated and will
```

File: ChangeLog [hunk 3]
```diff
@@ -40,6 +40,10 @@ Release date: TBA
...
+  be removed in 3.0.0. They are complex meta-inference functions that are not
+  suitable for astroid.
```
Comment (cdce8p) [inline ChangeLog:45]:
```suggestion
  be removed in 3.0.0. They are complex meta-inference functions that are better
  in pylint. Import them from ``pylint.checkers.utils`` instead
  (requires pylint ``2.12``).
```
Maybe something like this?

**Comment by DanielNoord** [comment]
I think referring to the changelog/`pylint` in the actual warning would be user-friendly!

**Pull request merged**